src/network/networkd-dhcp4.c

```
@@ -1141,6 +1144,219 @@ static int dhcp_lease_acquired(sd_dhcp_client *client, Link *link) {
+                return r;
+        }
+
+        sd_dhcp_lease_unref(link->dhcp_lease);
+        link->dhcp_lease = TAKE_PTR(lease);
+        link_dirty(link);
+
+        log_link_info(link, "Loaded saved DHCP lease");
+
+        return 1;
+}
+
+static bool dhcp_client_persist_leases(Link *link) {
+        assert(link);
+        assert(link->manager);
+        assert(link->network);
+
+        if (link->network->dhcp_client_persist_leases >= 0)
+                return link->network->dhcp_client_persist_leases > 0;
+
+        return link->manager->dhcp_client_persist_leases > 0;
+}
+
+int link_get_dhcp_client_lease_path(Link *link, int *ret_dir_fd, char **ret_path) {
+        assert(link);
+        assert(link->network);
+        assert(ret_dir_fd);
+
+        /* This does not copy fd. Do not close fd stored in ret_dir_fd. */
+
+        if (link->network->dhcp_client_persist_leases == DHCP_CLIENT_PERSIST_LEASES_NO) {
+                *ret_dir_fd = -EBADF;
+                *ret_path = NULL;
+                return 0;
+        }
+
+        if (link->manager->persistent_storage_fd < 0)
+                return -EBUSY;
+
+        _cleanup_free_ char *p = path_join("leases", link->ifname);
+        if (!p)
+                return -ENOMEM;
+
+        *ret_dir_fd = link->manager->persistent_storage_fd;
+        *ret_path = TAKE_PTR(p);
+        return 1;
+}
+
+static int dhcp_configure_with_persistent_lease(Link* link) {
+
+        _cleanup_free_ char *lease_file = NULL;
+        int dir_fd;
+        int r;
+
+        r = link_get_dhcp_client_lease_path(link, &dir_fd, &lease_file);
+
+        if (r < 0) {
+                log_link_debug_errno(link, r, "Failed to get persistent lease path, ignoring: %m");
+                return r;
+        }
+
+        if (r > 0) {
+                r = sd_dhcp_client_set_lease_file(link->dhcp_client, dir_fd, lease_file);
+                if (r < 0)
+                        log_link_warning_errno(link, r, "Failed to set lease file: %m");
```

Why error is not propagated?